sdk/identity/identity/src/msal/types.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as msalCommon from "@azure/msal-common";
+
+/**
+ * The shape we use return the token (and the expiration date).
+ * @internal
+ */
+export interface MsalToken {
+  accessToken?: string;
+  expiresOn: Date | null;
+}
+
+/**
+ * Represents the common properties of any of the MSAL responses.
+ * @internal
+ */
+export interface MsalResult {
+  account: msalCommon.AccountInfo | null;
+  accessToken: string;
+  expiresOn: Date | null;
+}
+
+/**
+ * The record to use to find the cached tokens in the cache.
+ */
+export interface AuthenticationRecord {
```

Comment:
I don't see the serialization methods on this type. How are users intended to serialize / deserialize it?

Reply:
They can JSON.stringify it 🤔 that would be pretty common on JS. But what do you think @xirzec , @bterlson ?

Reply:
We need to make sure this serializes the same as other languages as `AuthenticationRecord` should be sharable between applications in different languages.

Reply:
Alright, I'll do this! Do you have a snippet of how the format should be?

Reply:
I'll add tests too.

Reply:
Python serializes to string with properties with these names: https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/identity/azure-identity/tests/test_auth_record.py#L13

I'll use that then.

Reply:
So here's what I did:

`AuthenticationRecord` can be serialized into string with its property `serialize()`. To deserialize from string into an `AuthenticationRecord`, users can use the new function `deserializeAuthenticationRecord()`.

- Here's the `deserializeAuthenticationRecord()` source: https://github.com/Azure/azure-sdk-for-js/pull/14064/files#diff-45b41baedec5c607eb7d96a259c50b96e80c2737b9528f693c7248b504919336R48
- I added some small but relevant documentation on the `serialize()` method of the `AuthenticationRecord`: https://github.com/Azure/azure-sdk-for-js/pull/14064/files#diff-aa46e35bcc3d895662158885248c668eb48b19a1fffce4a986b6397a30b5defaR73
- I also added one test: https://github.com/Azure/azure-sdk-for-js/pull/14064/files#diff-8faccfec2f35c56ca7021b4b87709dd257e55fd5c248747a05a1eb153ec368deR183